src/components/pages/userCoursesListPage.tsx

```diff
@@ -1,7 +1,99 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { UserCourse, Course } from 'devu-shared-modules'
+
 
 import PageWrapper from 'components/shared/layouts/pageWrapper'
+import UserCourseListItem from 'components/listItems/userCourseListItem'
+import Dropdown from 'components/shared/inputs/dropdown'
+
+import RequestService from 'services/request.service'
+
+import styles from './userCoursesListPage.scss'
+import LoadingOverlay from 'components/shared/loaders/loadingOverlay'
+
+const filterOptions = [
+  { label: 'All', value: 'all' },
+  { label: 'Active', value: 'active' },
+  { label: 'Inactive', value: 'inactive' },
+  { label: 'Dropped', value: 'dropped' },
+]
+
+const UserCoursesListPage = () => {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const [userCourses, setUserCourses] = useState(new Array<UserCourse>())
+  const [courses, setCourses] = useState<Record<string, Course>>({})
+  const currentFilter = JSON.parse(localStorage.getItem('status') || '')
+  const [filter, setFilter] = useState<'all' | 'active' | 'inactive' | 'dropped'>(currentFilter)
+
+  const fetchData = async () => {
+    try {
+      // The filter isn't implemented by the API yet
+      const userCourses: UserCourse[] = await RequestService.get(`/api/user-courses?filterBy=${filter}`)
+      const courseRequests = userCourses.map((u) => RequestService.get(`/api/courses/${u.courseId}`))
+      const courses: Course[] = await Promise.all(courseRequests)
+
+      localStorage.setItem('status', JSON.stringify(filter) || '')
+
+      // Mapify course ids so we can look them up more easilly via their id
+      const courseMap: Record<string, Course> = {}
+      for (const course of courses) courseMap[course.id || ''] = course
+
+      setUserCourses(userCourses)
+      setCourses(courseMap)
+    } catch (error) {
+      setError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [filter])
+
+  if (loading) return <LoadingOverlay />
+
+  type Props = {
+    error: Error | null
+  }
+
+  const ErrorPage = ({error}: Props) => {
```

Comment:
Oh I'm sorry I wasn't clear. Just make this it's own file. Include the page wrapper in the return statement here. And just call the file errorPage and toss it in the pages directory. 

Also, bonus: you can make this function without an  explicit return to save the nesting

```javascript
const ErrorPage = ({ error }: Props => (
  <PageWrapper>
    // content here
  </PageWrapper>
)
```

Reply:
Should we consider placing it in the `src > shared > errors` folder, or is it too circumstantial? Just double-checking to be sure.

Reply:
Because it's the full page I'd probably put it in pages.

It's possible that we should make an error component that is then called by the errorpage component, but that feels like a bit much for this. If we find we want to display error mesasges elsewhere, we can think about that then